packages/components/src/popover/index.js

```diff
@@ -80,24 +80,38 @@ class Popover extends Component {
 	}
 
 	componentWillUnmount() {
-		this.toggleWindowEvents( false );
-
 		clearTimeout( this.focusTimeout );
+		this.toggleAutoRefresh( false );
 	}
 
-	toggleWindowEvents( isListening ) {
-		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+	toggleAutoRefresh( isActive ) {
+		// Refresh the popover every time the window is resized or scrolled
+		const handler = isActive ? 'addEventListener' : 'removeEventListener';
+		window[ handler ]( 'resize', this.throttledRefresh );
+		window[ handler ]( 'scroll', this.throttledRefresh, true );
 
-		window.cancelAnimationFrame( this.rafHandle );
-		window[ handler ]( 'resize', this.throttledComputePopoverPosition );
-		window[ handler ]( 'scroll', this.throttledComputePopoverPosition, true );
+		/*
+		 * There are sometimes we need to reposition or resize the popover that are not
+		 * handled by the resize/scroll window events (i.e. CSS changes in the layout
+		 * that changes the position of the anchor).
+		 *
+		 * For these situations, we refresh the popover every 0.5s
+		 */
+		if ( isActive ) {
+			this.autoRefresh = setInterval( () => {
```

Comment:
Do we need the wrapping function?

vs.

```js
setInterval( this.throttledRefresh, 500 )
```

Reply:
you're right, not needed :)